### PR TITLE
update code for nom 7.1.1

### PR DESCRIPTION
**Pull request by phill-85**
I've updated this to nom 7.1.1. In the process I've added some more tests.

A lot of the custom error code is commented out for the time being. I was thinking of using the nom VerboseError type. I would appreciate your input on this.

### CODE REVIEW

**Comment by compenguy** [comment]
In general, it looks like a pretty clean update to the new nom syntax.

As for using VerboseError, I'm thinking I'd be ok with that being the default, but maybe make it selectable between nom::error::VerboseError and nom::error::Error?  I think it would just be a matter of adding a feature called "verbose-error" (or similar) to Cargo.toml and adding it to the default features list, then doing something like this in code:

```
#[cfg(verbose-error)]
import nom::error::VerboseError as NomError;
#[cfg(not(verbose-error)]
import nom::error::Error as NomError;
```

I'm only suggesting this because VerboseErrors are much more informative, but it also looks like (judging from the docs) it takes extra work to format the error in a nicely readable way.  But I mostly view that as a nice to have, and I could be talked out of it. The VerboseError is still an overall improvement over what I had before, I think. Whichever way you decide, when you update the pull request with error handling I'd like to see an example of what a parser error looks like.

All in all, though, I'm really happy with what I see, and feel free to take the PR out of draft state when you re-add error handling, however you choose to do it.

**Comment by phill-85** [comment]
I meant to say that I'm still a newb to rust, and an amateur programmer. So any other comments you have will be gratefully accepted!

**Comment by compenguy** [comment]
I noticed that an RTF file referenced the the tokenizer.rs tests is missing from the project.  If you'd like to add it in, it can be found at http://www.snake.net/software/RTF/RTF-Spec-1.7.rtf .

Also, if you wouldn't mind adding documentation for the API, you can get compiler guidance as to what needs documenting by adding `#![deny(missing_docs)]` to the top of lib.rs.

Otherwise, once `cargo check` and `cargo clippy` come up clean, I think you're good to go.

**Comment by phill-85** [comment]
> I noticed that an RTF file referenced the the tokenizer.rs tests is missing from the project. If you'd like to add it in, it can be found at http://www.snake.net/software/RTF/RTF-Spec-1.7.rtf .

I noticed that too, but I had assumed that you hadn't added it because of copyright issues.....

**Comment by compenguy** [comment]
> I noticed that too, but I had assumed that you hadn't added it because of copyright issues.....

That was probably why I didn't initially include it.  But the [US Library of Congress's digital format entry for RTF](
https://www.loc.gov/preservation/digital/formats/fdd/fdd000473.shtml) (which I had not seen before until this week) says:

> Microsoft states in the latest RTF specification, "RTF serves as both a standard of data transfer between word processing software, document formatting, and a means of migrating content from one operating system to another." The compilers of this resource infer from this that, although there may be patents associated with features described in the RTF specification, Microsoft has encouraged the use of the specification to build tools that can read or write RTF files. Microsoft does claim copyright, with all rights reserved, on the specification document, but has not objected to the copies of old versions archived by tool developers.

**Comment by phill-85** [comment]
> As for using VerboseError, I'm thinking I'd be ok with that being the default, but maybe make it selectable between nom::error::VerboseError and nom::error::Error? I think it would just be a matter of adding a feature called "verbose-error" (or similar) to Cargo.toml and adding it to the default features list, then doing something like this in code:

> I'm only suggesting this because VerboseErrors are much more informative, but it also looks like (judging from the docs) it takes extra work to format the error in a nicely readable way. But I mostly view that as a nice to have, and I could be talked out of it. The VerboseError is still an overall improvement over what I had before, I think. Whichever way you decide, when you update the pull request with error handling I'd like to see an example of what a parser error looks like.

It appears that Verbose error may not be very useful in this case. I came to this conclusion because we use alt right at the beginning. (src/tokenizer.rs line 187) Which try each parser, and if all of them produces an error returns the error of the last parser it tried. So if the first branch is the correct one, and parses 99% but falls down on the last 1% the last parser in the alt will be reported as the failure, even if it matched 0%.....

**Comment by compenguy** [comment]
> It appears that Verbose error may not be very useful in this case.

Well, that makes it easy.  Thanks for looking into that.

**Comment by compenguy** [comment]
I was doing some cleanups on my rtf2text cli utility, so I wound up making some minor changes to rtf-grimoire in the process.

* made a minor change to the ParseError struct to make it impl std::error::Error
* added the RTF spec doc

File: src/raw.rs [hunk 1]
```diff
@@ -7,22 +7,20 @@
 //
 
 use std;
-use nom;
+use std::result::Result::Err;
```
Comment (compenguy) [inline src/raw.rs:10]: I don't think this should be required - I think this is imported by the default prelude.  `Err(e)` has always just worked for me without a `use` statement, anyway.

Just to be sure, you should verify that `cargo check` and `cargo clippy` both come up clean.

**Comment by phill-85** [comment]
Sorry I've not had much time of late to complete this, but I do plan to
continue this work when I get the time.

**Comment by compenguy** [comment]
@mike-kfed has picked up this change in #2, building on top of the work here, so I'm going to close this PR in favor the work done there.

**Pull request closed without merging**